scripts/tundra/tools/msvc-vscommon-next.lua

```diff
@@ -0,0 +1,204 @@
+-- for usage with Visual Studio 2017 and later, 
+-- does not work with earlier versions of Visual Studio
+
+module(..., package.seeall)
+
+local native = require "tundra.native"
+local native_path = require "tundra.native.path"
+
+local vs_products = {
+  "BuildTools", --default
+  "Community",
+  "Professional",
+  "Enterprise"
+}
+
+local function find_vc_tools(installation_path, version, vs_product)
+  local path, stat
+
+  path = native_path.join(installation_path, version)
+  path = native_path.join(path, vs_product)
+  path = native_path.join(path, "VC\\Auxiliary\\Build\\vcvarsall.bat")
+
+  stat = native.stat_file(path)
+  if stat.exists then
+    return path
+  end
+  return nil
+end
+
+local supported_arch_mappings = {
+  ["x86"] = "x86",
+  ["amd64"] = "amd64",
+  ["arm"] = "arm",
+  ["arm64"] = "arm64",
+  
+  --alias
+  ["x64"] = "amd64", 
+}
+
+local supported_arch_tuples = {
+  ["x86"] = true,
+  ["amd64"] = true,
+  ["x86_amd64"] = true,
+  ["x86_arm"] = true,
+  ["x86_arm64"] = true,
+  ["amd64_x86"] = true,
+  ["amd64_arm"] = true,
+  ["amd64_arm64"] = true,
+}
+
+local function get_arch(arch)
+  local arch2 = supported_arch_mappings[arch:lower()] 
+  return arch2
+end
+
+-- dump keys of table into sorted array
+local function keys(tbl)
+  local ks, i = {}, 1
+  for k, _ in pairs(tbl) do
+    ks[i] = k
+    i = i + 1
+  end
+  table.sort(ks)
+  return ks
+end
+
+local function get_arch_tuple(host_arch, target_arch)
+  local host_arch2 = get_arch(host_arch)
+  local target_arch2 = get_arch(target_arch)
+
+  if host_arch2 == nil then
+    error("unknown host architecture '" .. host_arch .. "' expected one of " .. table.concat(keys(supported_arch_mappings), ", "))
+  end
+  if target_arch2 == nil then
+    error("unknown target architecture '" .. target_arch .. "' expected one of " .. table.concat(keys(supported_arch_mappings), ", "))
+  end
+
+  if host_arch2 == target_arch2 then
+    return host_arch2
+  end
+  return host_arch2 .. "_" .. target_arch2
+end
+
+function apply_msvc_visual_studio(version, env, options)
+  if native.host_platform ~= "windows" then
+    error("the msvc toolset only works on windows hosts")
+  end
+  
+  tundra.unitgen.load_toolset('msvc', env)
+
+  -- this variable will be applicable to versions after Visual Studio 2017 as well
+  -- env:set("TUNDRA_VS2017_ROOT", "C:\\Program Files (x86)\\Microsoft Visual Studio\\" .. version)
+
+  options = options or {}
+
+  local target_arch = options.TargetArch or "x64"
+  local host_arch = options.HostArch or "x64"
+  local arch_tuple = get_arch_tuple(host_arch, target_arch)
+  local platform_type = options.PlatformType --default empty
+  local sdk_version = options.SdkVersion --default from vcvarsall.bat
+  local installation_path = options.InstallationPath or "C:\\Program Files (x86)\\Microsoft Visual Studio"
```

Comment:
I could figure out how to actually pass an env var for the bootstrapping of the tooling. It would appear as if the env var is not made available here so I opted to use the option `InstallationPath`.